File: ts/twoSum.ts
const twoSum = (nums: number[], target: number): number[] => {
  const indices = new Map();
  return nums.reduce<number[]>((result, number, index) => {
    const remainder = target - number;
    const storedIndex = indices.get(remainder);
    if (storedIndex >= 0) {
      return [storedIndex, index];
    } else {
      indices.set(number, index);
    }
    return result;
  }, []);
};

File: rs/twoSum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut index_hashmap = HashMap::with_capacity(nums.len());

        for (idx, &n) in nums.iter().enumerate() {
            let y = target - n;
            if let Some(&i) = index_hashmap.get(&y) {
                return vec![i as i32, idx as i32];
            } else {
                index_hashmap.insert(n, idx);
            }
        }

        vec![]
    }
}
